### PR TITLE
Deserialize sb3 assets before loading data

### PR DESCRIPTION
**Pull request by mzgoddard**
### Resolves

Load faster.

### Proposed Changes

Depends on #1972

Parse and load costumes and sounds before parsing the rest of the data.

### Reason for Changes

Parsing a project's data and creating the relevant Sprites, Blocks, and other objects takes a fair bit of time. We can load the overall project faster by making the network requests for the necessary data and then parsing the whole data set.

### CODE REVIEW

File: test/unit/project_changed_state.js [hunk 1]
```diff
@@ -48,7 +48,11 @@ test('Adding a sprite (from sprite3) should emit a project changed event', t =>
     vm.addSprite(sprite3).then(() => {
         t.equal(projectChanged, true);
         t.end();
-    });
+    })
+        .catch(e => {
```
Comment (kchadha) [inline test/unit/project_changed_state.js]: Why is this change necessary?

**Review by kchadha — APPROVED** [review]
Small question in the comments, but otherwise LG!

File: test/unit/project_changed_state.js (hunk 1 above)
Comment (mzgoddard) [reply test/unit/project_changed_state.js]: Hmm. This isn't necessary. I added it to clarify why the test was failing because an error in the chain would keep the above then clause from executing. The lack of an error report would have made it hard to fix. But now that I fixed that issue we don't need this.

Should I remove it?

Comment (mzgoddard) [reply test/unit/project_changed_state.js]: Wrote this last week. But its posting was stucking a review queue. I hit the wrong button.

Comment (mzgoddard) [reply test/unit/project_changed_state.js]: Gonna remove this catch statement.

Comment (kchadha) [reply test/unit/project_changed_state.js]: LG! Thanks!

**Review by kchadha — APPROVED** [review]
LGTM!
